### PR TITLE
Improve cross-compilation support for arm and arm64

### PR DESCRIPTION
**Pull request by manu-silicon**
Merge parts for PR #6194 which were not included in PR #6585 so that cross compilation scripts are similar to the ones in CoreCLR and CoreRT.

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @leemgs, @benpye, @jkotas

File: cross/arm/tryrun.cmake [hunk 1]
```diff
@@ -1,6 +1,7 @@
 SET( HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP_EXITCODE
      0
      CACHE STRING "Result from TRY_RUN" FORCE)
+
```
Comment (benpye) [inline cross/arm/tryrun.cmake]: Was this intentional, looks like a mistake.

File: cross/arm64/tryrun.cmake [hunk 2]
```diff
@@ -1,6 +1,7 @@
 SET( HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP_EXITCODE
      0
      CACHE STRING "Result from TRY_RUN" FORCE)
+
```
Comment (benpye) [inline cross/arm64/tryrun.cmake]: Same as the above.

**Comment by benpye** [comment]
LGTM modulo minor comments, won't be able to check it this weekend though.

File: cross/arm/tryrun.cmake (hunk 1 above)
Comment (manu-silicon) [inline cross/arm/tryrun.cmake]: Yes, to be like CoreCLR and CoreRT that have empty lines between the various SET commands.

Comment (benpye) [inline cross/arm/tryrun.cmake]: Yeah my bad sorry, saw a + and assumed it was a literal + forgetting it's used to mean addition.

**Comment by stephentoub** [comment]
@manu-silicon, checking in on this.  Is this PR still necessary?  Are you planning to address the feedback that was left?

**Comment by manu-silicon** [comment]
Unless what is currently in CoreFX enables this, I would still think it is relevant. However, it has been a while I worked on the ARM side and haven't tested this recently. Maybe @leemgs can comment on this.

**Comment by stephentoub** [comment]
@leemgs, @prajwal-aithal, could you comment on this and what you think the next steps are here? Thanks!

**Comment by prajwal-aithal** [comment]
@stephentoub @manu-silicon I agree that this PR brings consistency between cross build scripts in coreclr and corefx.

The current PR has merge conflicts, and thus they should be fixed first. In the changes made currently, the changes in the file `cross/build-rootfs.sh` need not be made in the newer version of the PR as this file has been re-organized in a previous PR. The other changes have not been done, and they can be included after the merge conflicts are resolved.

**Comment by Petermarcu** [comment]
It would be great to have this PR come to a conclusion. Should we close it and track it with an issue until someone has time to finish it up or will someone be able to get to it in the next couple days?

**Comment by manu-silicon** [comment]
I've updated the PR to match master. Not sure what the  Linux ARM Emulator failed in Release mode though.

**Comment by leemgs** [comment]
> Not sure what the Linux ARM Emulator failed in Release mode though.

/CC: @sjsinju, @prajwal-aithal

**Comment by Petermarcu** [comment]
@dotnet-bot test Linux ARM Emulator Release

**Comment by Petermarcu** [comment]
@manu-silicon , looks like there are still merge conflicts with master. Did you need some help understanding the CI failure? Who can help with that?

**Comment by Petermarcu** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator Release Cross Build please

**Comment by manu-silicon** [comment]
I'll check the conflicts and if anyone has an idea about the failure ```umount: /opt/linux-arm-emulator-root: device is busy.``` let me know.

**Comment by stephentoub** [comment]
I believe the "device is busy" error should have been fixed.  You may need to close this PR and re-open it (or open a new one) and rebase to pick up the revised netci.groovy changes that cause the legs to be produced.

**Comment by Petermarcu** [comment]
@manu-silicon, can you close this and submit a new rebased PR with the changes to pick up the Jenkins fix as Stephen suggests? Thanks for continuing to push this through!

**Comment by manu-silicon** [comment]
There are now too many changes and I need to check how things are done before I can re-apply my changes. I'm closing this now and when I'll have more time I'll re-open if there is a need to.

**Pull request closed without merging**